File: cpp/design-patterns/factory/naive/services_naive.cpp
void OrderService::notify(const User& user, const std::string& msg) {
    // ❌ the "which class?" decision is made right here, inline
    std::unique_ptr<Notifier> notifier = user.prefersSms
        ? std::unique_ptr<Notifier>(new SmsNotifier())
        : std::unique_ptr<Notifier>(new EmailNotifier());
    notifier->send(msg);
}

void BillingService::remind(const User& user, const std::string& msg) {
    // ❌ ...and copy-pasted again here
    std::unique_ptr<Notifier> notifier = user.prefersSms
        ? std::unique_ptr<Notifier>(new SmsNotifier())
        : std::unique_ptr<Notifier>(new EmailNotifier());
    notifier->send(msg);
}

File: typescript/design-patterns/factory/naive/services-naive.ts
class OrderService {
  notify(user: User, msg: string) {
    // ❌ the "which class?" decision is made right here, inline
    const notifier = user.prefersSms
      ? new SmsNotifier()
      : new EmailNotifier();
    notifier.send(msg);
  }
}

class BillingService {
  remind(user: User, msg: string) {
    // ❌ ...and copy-pasted again here, a little differently
    const notifier = user.prefersSms ? new SmsNotifier() : new EmailNotifier();
    notifier.send(msg);
  }
}
// Adding a Slack channel means editing BOTH of these — and every other
// service that ever constructed a notifier.